lib/streamlit/runtime/media_file_manager.py

```diff
@@ -56,106 +38,26 @@ def _get_session_id() -> str:
-    mimetype : str
-        Any string. Will be converted to bytes and used to compute a hash.
-        None will be converted to empty string.  [default: None]
-    file_name : str
-        Any string. Will be converted to bytes and used to compute a hash.
-        None will be converted to empty string. [default: None]
-    """
-    filehash = hashlib.new("sha224")
-    filehash.update(data)
-    filehash.update(bytes(mimetype.encode()))
-
-    if file_name is not None:
-        filehash.update(bytes(file_name.encode()))
-
-    return filehash.hexdigest()
-
-
-def _get_extension_for_mimetype(mimetype: str) -> str:
-    # Python mimetypes preference was changed in Python versions, so we specify
-    # a preference first and let Python's mimetypes library guess the rest.
-    # See https://bugs.python.org/issue4963
-    #
-    # Note: Removing Python 3.6 support would likely eliminate this code
-    if mimetype in PREFERRED_MIMETYPE_EXTENSION_MAP:
-        return PREFERRED_MIMETYPE_EXTENSION_MAP[mimetype]
-
-    extension = mimetypes.guess_extension(mimetype)
-    if extension is None:
-        return ""
-
-    return extension
+class MediaFileMetadata:
+    """Metadata that the MediaFileManager needs for each file it manages."""
 
-
-class MediaFile:
-    """Abstraction for file objects."""
-
-    def __init__(
-        self,
-        file_id: str,
-        content: bytes,
-        mimetype: str,
-        file_name: Optional[str] = None,
-        file_type: MediaFileType = MediaFileType.MEDIA,
-    ):
-        self._file_id = file_id
-        self._content = content
-        self._mimetype = mimetype
-        self._file_name = file_name
-        self._file_type = file_type
+    def __init__(self, kind: MediaFileKind = MediaFileKind.MEDIA):
+        self._kind = kind
         self._is_marked_for_delete = False
 
-    def __repr__(self) -> str:
-        return util.repr_(self)
-
-    @property
-    def url(self) -> str:
-        return f"{STATIC_MEDIA_ENDPOINT}/{self.id}{self.extension}"
-
-    @property
-    def id(self) -> str:
-        return self._file_id
-
     @property
-    def extension(self) -> str:
-        return _get_extension_for_mimetype(self.mimetype)
+    def kind(self) -> MediaFileKind:
+        return self._kind
 
     @property
-    def content(self) -> bytes:
-        return self._content
+    def is_marked_for_delete(self) -> bool:
+        return self._is_marked_for_delete
 
-    @property
-    def mimetype(self) -> str:
-        return self._mimetype
-
-    @property
-    def content_size(self) -> int:
-        return len(self._content)
-
-    @property
-    def file_type(self) -> MediaFileType:
-        return self._file_type
-
-    @property
-    def file_name(self) -> Optional[str]:
-        return self._file_name
-
-    def _mark_for_delete(self) -> None:
+    def mark_for_delete(self) -> None:
         self._is_marked_for_delete = True
 
 
-class MediaFileManager(CacheStatsProvider):
```

Comment:
MediaFileManager no longer stores files, so it is not a CacheStatsProvider. (But MemoryMediaFileManager is!)